Chung/week_6/2_inheritance/stations_challenge.py

```diff
@@ -0,0 +1,113 @@
+        super().__init__(station_name, location)
+        self.lines = lines
+
+    def show_info(self):
+        print(f'{self.station_name} station is located at {self.location}.',
+        f'The lines that stop at the station are {", ".join(self.lines)}.', sep='\n')
+
+print('Question 2: Make an example subway station')
+'''
+Using your SubwayStation class, instantiate a subway station with the info below.
+Then run the show_info() method to make sure you get the station_name, location, and lines printed out
+
+station_name: '14th Street'
+location: '14th street and 7th avenue'
+lines: ['1', '2', '3', 'L']
+'''
+sub1_station = SubwayStation('14th street', '14th street and 7th avenue', ['1', '2', '3', 'L'])
+sub1_station.show_info()
+
+print('Question 3: Making the BusStation Class')
+
+'''
+Using the Station class below as the parent, make a child class called BusStation
+
+BusStation should:
+-have an additional attribute called 'routes' that is user-defined as a list during initialization.
+    this will indicate where buses can go from this station. For example, ['DC', 'Philly', 'Pittsburgh']
+-have an additional attribute called 'open' that is always initalized as True (a boolean variable)
+-have additional methods called open_station() and close_station() to open and close the station
+-override the show_info() method from Station to display the bus routes and if the station is open, in addition to the station name and location
+'''
+
+class BusStation(Station):
+    def __init__(self, station_name, location, routes):
+        super().__init__(station_name, location)
+        self.routes = routes
+        self.open = open
+
+    def open_station(self):
+        self.open = True
+
+    def close_station(self):
+        self.open = False
+
+    def show_info(self):
+        print(f'{self.station_name} station is located at {self.location}.',
+        f'The routes that the station go through are {", ".join(self.routes)}.',
+        f'The station is now {"open" if self.open else "closed"}.', sep='\n')
+
+print('Question 4: Make an example bus station')
+'''
+Using your BusStation class, instantiate a bus station with the info below.
+Then, run the show_info() method to make sure you get the station_name, location, routes, and whether the station is open printed out
+Then, demonstrate that you can close and open the bus station
+    i.e. that the show_info() method reflects correctly when the station is open versus closed
+
+station_name: 'NYC Megabus Stop'
+location: '34th street and 12th avenue'
+lines: ['Boston', 'DC', 'Philly']
+'''
+
+nyc_megabus_stop = BusStation('NYC Megabus Stop', '34th street and 12th avenue', ['Boston', 'DC', 'Philly'])
+
+nyc_megabus_stop.show_info()
+nyc_megabus_stop.close_station()
+nyc_megabus_stop.show_info()
+
+print('Question 5: Importing your classes')
+
+
+'''
+Now, it's time to design a few more stations of your own in another script!
+
+Make a new python script called "station_planning.py"
```

Comment:
Did you make the script and push it? I don't see it in this request.

Reply:
I don't know what you mean. I see `commits (2)` and `files changed (2)` on the tabs for this PR, and I see the `station_planning.py` file when I click on the `files changed (2)` tab and choose the first commit after clicking on the `commits (2)` tab.